tests/scripts/setup-pytest-env.sh

```diff
@@ -32,7 +32,33 @@ export PYTHONPATH="${TVM_PATH}/python"
 export TVM_PYTEST_RESULT_DIR="${TVM_PATH}/build/pytest-results"
 mkdir -p "${TVM_PYTEST_RESULT_DIR}"
 
+if [ -n "${CI_PYTEST_NUM_CPUS-}" ]; then
+    PYTEST_NUM_CPUS=${CI_PYTEST_NUM_CPUS}
+else
+    PYTEST_NUM_CPUS=$(nproc)
+    if [ -z "${PYTEST_NUM_CPUS}" ]; then
+        PYTEST_NUM_CPUS=1
+    elif [ ${PYTEST_NUM_CPUS} -gt 1 ]; then
+        PYTEST_NUM_CPUS=$(expr ${PYTEST_NUM_CPUS} - 1)  # Don't nuke interactive work.
+    fi
+fi
+
+if [ ${PYTEST_NUM_CPUS} -gt 8 ]; then
+    PYTEST_NUM_CPUS=8  # It usually doesn't make sense to launch > 8 workers
+fi
+
+# DNS: remove after we actually fix up CI_PYTEST_NUM_CPUS in Jenkinsfile
+if [ ${PYTEST_NUM_CPUS} -gt 2 ]; then
+    PYTEST_NUM_CPUS=2  # Fix to 2 CPUs for Jenkins
+fi
+
+
 function run_pytest() {
+    local extra_args=( )
+    if [ "$1" == "--parallel" ]; then
```

Comment:
Since the documentation [currently recommends](https://github.com/apache/tvm/blob/main/docs/contribute/pull_request.rst#L121) using `task_python_unittest.sh` to run the unit tests, we should either make sure that location also has `pytest-xdist` in the list of packages to install in order to run the tests, or check whether `pytest-xdist` is installed by wrapping this in `if python3 -c "import xdist" > /dev/null 2>&1; then`.